src/neo/Network/P2P/TaskManager.cs

```diff
@@ -169,12 +169,13 @@ private void OnRegister(VersionPayload version)
             RequestTasks();
         }
 
-        private void OnUpdate(uint lastBlockIndex)
+        private void OnUpdate(Update update)
         {
             if (!sessions.TryGetValue(Sender, out TaskSession session))
                 return;
-            session.LastBlockIndex = lastBlockIndex;
-            RequestTasks();
+            session.LastBlockIndex = update.LastBlockIndex;
+            session.ExpireTime = TimeProvider.Current.UtcNow.AddMilliseconds(PingCoolingOffPeriod);
+            if (update.RequestTasks) RequestTasks();
```

Comment:
It will call `RequestTasks()` in `OnBlock()`. So we don't need to call it again.

Reply:
That's the reason why I sent `false` in `OnBlock`. Do you think that also we don't need it when `OnPing` ?

Reply:
I believe that @shargon current change is to send False when it comes generic from IIventory.